backend/inspirehep/records/tasks.py

```diff
@@ -43,3 +54,66 @@ def update_records_relations(uuids):
 
     db.session.commit()
     return uuids
+
+
+@shared_task(
+    bind=True,
+    queue="indexer_task",
+    acks_late=True,
+    retry_backoff=True,
+    retry_kwargs={"max_retries": 3},
+    autoretry_for=(OperationalError, TransportError),
+)
+def redirect_references_to_merged_record(self, uuid):
+    record = InspireRecord.get_record(uuid)
+    deleted_record_refs = record.get_new_deleted_records()
+    if deleted_record_refs:
+        record_ref = get_value(record, "self.$ref")
```

Comment:
you need this to be available, otherwise things will fail, and it should always be there, so better do
```suggestion
        record_ref = record["self"][$ref"]
```
which is 1) slightly more performant and 2) fails early and loudly in case it's not there.